geode-core/src/main/java/org/apache/geode/internal/cache/execute/InternalFunctionExecutionServiceImpl.java

```diff
@@ -106,24 +106,32 @@ public Execution onMember(String... groups) {
     return onMember(getDistributedSystem(), groups);
   }
 
+  protected Pool findPool(String poolName) {
+    return PoolManager.find(poolName);
+  }
+
   @Override
   public Execution onRegion(Region region) {
     if (region == null) {
-      throw new FunctionException(
-          String.format("%s passed is null", "Region instance "));
+      throw new FunctionException("Region instance passed is null");
     }
 
     ProxyCache proxyCache = null;
     String poolName = region.getAttributes().getPoolName();
     if (poolName != null) {
-      Pool pool = PoolManager.find(poolName);
-      if (pool.getMultiuserAuthentication()) {
-        if (region instanceof ProxyRegion) {
-          ProxyRegion proxyRegion = (ProxyRegion) region;
-          region = proxyRegion.getRealRegion();
-          proxyCache = proxyRegion.getAuthenticatedCache();
-        } else {
-          throw new UnsupportedOperationException();
+      Pool pool = findPool(poolName);
+
+      if (pool == null) {
+        throw new IllegalStateException(String.format("Could not find a pool named %s.", poolName));
```

Comment:
I second Jason in that I am super happy that you added a bunch of unit tests here!  Just curious, what happens in production if an IllegalStateException is thrown here?  Where does this exception bubble up to in production and what is the consequence?  Do we have any DUnit coverage for this case which might better exercise the whole system?

Reply:
I missed @pivotal-jbarrett's comments above which I think express a similar sentiment.

Reply:
@mcmellawatt: see [here](https://github.com/apache/geode/pull/3569#issuecomment-491105692). The exception will bubble up to the client application, which is what we do when we can't find the `pool` in similar scenarios:

- [GemFireCacheImpl.getQueryService(String poolName)](https://github.com/apache/geode/blob/develop/geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java#L4635-L4642).
- [DynamicRegionFactory.doInternalInit(InternalCache theCache)](https://github.com/apache/geode/blob/develop/geode-core/src/main/java/org/apache/geode/cache/DynamicRegionFactory.java#L266-L270).

I will try to write an integration/distributed test to bypass the new validation and add it to this PR, don't know if it's feasible, though, as the `PoolManagerImpl` class is specifically implemented to prevent inheritance.